### PR TITLE
Have animations more closely match the HTML spec

### LINKED ISSUE #13865: Fix animation processing ordering and reenable animations/basic-transition.html test
In #12623, we added some hacky machinery that causes us to mutate style on non-dirty nodes during restyle to tick animations. This causes all kinds of trouble for the new incremental restyle architecture, since we're not guaranteed to have mutable style data for non-dirty nodes (the style might be owned by the layout frame).

Emilio thinks the right approach is to mark the node as dirty when processing transitionend. However, that still doesn't cause us to pass the test, because Servo doesn't properly implement the processing model in the spec [1], and fires requestAnimationFrame callbacks _before_ animation events, rather than after.

Emilio is trying to fix the processing model a bit in #13864, but says that more refactoring will need to be done to pass the test. So we decided to disable the test for now to unblock things.

CC @emilio @notriddle

[1] https://html.spec.whatwg.org/#event-loop-processing-model:run-the-animation-frame-callbacks

### PR DESCRIPTION
**Pull request by mrobinson**
These two commits do two major things:

**Have animations ticks trigger a restyle**: This corrects synchronization issues with animations,
where the values used in layout are out of sync with what is returned by `getComputedStyle`.

**Tick the animation timer in script according to spec**: This greatly reduces the flakiness of
animation and transitions tests.

Fixes #13865.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #13865
- [x] There are tests for these changes

### CODE REVIEW

**Comment by jdm** [comment]
Should this PR enable the disabled tests, or are there other prerequisites?

**Comment by mrobinson** [comment]
@jdm, yep. Currently, I think there are only two disabled:
 * mozilla/css/animations/basic-transition.html
 * mozilla/css/animations/transition-raf.html

These are enabled by [removing the ini files](https://github.com/servo/servo/pull/26407/files#diff-3101d239acbaeb757c93c981b546d585) which were disabling them. Could it be that there are more disabled in another directory?

**Comment by mrobinson** [comment]
@bors-servo try=wpt

Removing two expectations for newly passing tests.

**Comment by jdm** [comment]
What about the animation-related test directories skipped in include.ini?

**Comment by mrobinson** [comment]
I wasn't sure if it was better to unskip intermittent tests later, but I'm happy to give it a shot in this change.

**Comment by mrobinson** [comment]
It looks like there is more required to eliminate the flakiness from animations and transitions tests. My suspicion is that this is #19242 and perhaps, in addition, making sure that transition and animation events arrive to the script thread before the next animation tick is triggered. I'll start taking a look at this tomorrow. It seems like it would be good to have this all sorted out before going too much further.

**Review by jdm — APPROVED** [review]

File: components/script/dom/node.rs [hunk 1]
```diff
@@ -170,13 +170,13 @@ bitflags! {
...
-        const CLICK_IN_PROGRESS = 1 << 2;
+        const CLICK_IN_PROGRESS = 1 << 3;
 
         #[doc = "Specifies whether this node is focusable and whether it is supposed \
                  to be reachable with using sequential focus navigation."]
-        const SEQUENTIALLY_FOCUSABLE = 1 << 3;
+        const SEQUENTIALLY_FOCUSABLE = 1 << 4;
 
-        // There are two free bits here.
+        // There is one bit free here.
```
[the remark this answers is not in the document]
Comment (mrobinson) [reply components/script/dom/node.rs:179]: Whoops. This is left over from a previous version of this change where I added `NodeFlags::HAS_ANIMATION_ONLY_DIRTY_DESCENDANTS`. It turns out that isn't necessary for this change, so I'll remove these changes as well.

**Comment by jdm** [comment]
I'm liking these changes more and more!

**Review by jdm — APPROVED** [review]

**Pull request merged**